lib/oauth/oauth.ex

```diff
@@ -25,3 +62,34 @@ defmodule OAuth do
     end
   end
 end
+
+defmodule OAuth.Cache do
+  use Agent
+
+  def start_link(initial_value) do
+    Agent.start_link(fn -> initial_value end, name: __MODULE__)
+  end
+
+  def get(get_fn) do
+    Agent.get(__MODULE__, get_fn)
+  end
+
+  def store(update_fn) do
+    Agent.update(__MODULE__, update_fn)
+  end
+end
+
+defmodule OAuth.Cache.Application do
+  @moduledoc false
+
+  use Application
+
+  def start(_type, _args) do
+    children = [
+      {OAuth.Cache, %{}}
+    ]
+
+    opts = [strategy: :one_for_one, name: Bootstrap.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
```

Comment:
Nice, you got it working 👍
Just a note:tThis should typically be The application of your project, not related to oauth specifically.
It is typically stored in `lib/project_name/application.ex`, where you should start your webserver etc as well.
This is not blocking though, you can always refactor it later.